### PR TITLE
More stuff

### PR DESCRIPTION
**Pull request by d3nwah**

### CODE REVIEW

File: Source/AliveLibAO/Abe.cpp [hunk 1]
```diff
@@ -2468,16 +2470,147 @@ void Abe::BulletDamage_4220B0(Bullet* pBullet)
...
+        {
+            break;
+        }
+
+        if (pObj->field_4_typeId == Types::eDoor_21)
+        {
+            auto pDoor = static_cast<Door*>(pObj);
+            PSX_RECT pRect = {};
+            VGetBoundingRect_418120(&pRect, 1);
+            PSX_RECT pRect2 = {};
```
Comment (paulsapps) [inline Source/AliveLibAO/Abe.cpp:2488]: doorRect/abeRect?

File: Source/AliveLibAO/Abe.cpp [hunk 2]
```diff
@@ -2468,16 +2470,147 @@ void Abe::BulletDamage_4220B0(Bullet* pBullet)
...
+    // Are we actually on a door?
+    Path_TLV* pDoorTlv = gMap_507BA8.TLV_Get_At_446260(
+        FP_GetExponent(field_A8_xpos),
+        FP_GetExponent(field_AC_ypos),
+        FP_GetExponent(field_A8_xpos),
+        FP_GetExponent(field_AC_ypos),
+        TlvTypes::Door_6
+    );
+
+    if (!pDoorTlv)
```
Comment (paulsapps) [inline Source/AliveLibAO/Abe.cpp:2527]: dunno if AE was this way but if (pDoorTlv && NearDoorIsOpen) { stuff return 1; } return 0; seems better?

File: Source/AliveLibAO/Abe.cpp [hunk 3]
```diff
@@ -2468,16 +2470,147 @@ void Abe::BulletDamage_4220B0(Bullet* pBullet)
...
 }
 
-__int16 Abe::MoveLiftUpOrDown_42F190(FP /*ySpeed*/)
+__int16 Abe::MoveLiftUpOrDown_42F190(FP ySpeed)
 {
-    NOT_IMPLEMENTED();
-    return 0;
+    auto pLiftPoint = static_cast<LiftPoint*>(field_F8_pLiftPoint);
+
+    if (pLiftPoint)
```
Comment (paulsapps) [inline Source/AliveLibAO/Abe.cpp:2549]: move this code to another function like how it is in AE

File: Source/AliveLibAO/Abe.cpp [hunk 4]
```diff
@@ -2468,16 +2470,147 @@ void Abe::BulletDamage_4220B0(Bullet* pBullet)
...
+        SetActiveCameraDelayedFromDir_401C90();
+    }
+
+    pLiftPoint->Move_435740(FP_FromInteger(0), ySpeed, 0);
+
+    if (gBeeInstanceCount_5076B0 && word_5076AC)
+    {
+        return eAbeStates::State_141_BeesStrugglingOnLift_42F390;
+    }
+    if (sControlledCharacter_50767C == (BaseAliveGameObject *) this &&
```
Comment (paulsapps) [inline Source/AliveLibAO/Abe.cpp:2566]: redundant cast

File: Source/AliveLibAO/Abe.cpp [hunk 5]
```diff
@@ -2468,16 +2470,147 @@ void Abe::BulletDamage_4220B0(Bullet* pBullet)
...
+        !(field_10_anim.field_4_flags.Get(AnimFlags::eBit18_IsLastFrame)) &&
```
Comment (paulsapps) [inline Source/AliveLibAO/Abe.cpp:2567]: extra brackets now its a function call, remove if you want

Comment (d3nwah) [reply Source/AliveLibAO/Abe.cpp:2567]: I don't want to

Comment (d3nwah) [reply Source/AliveLibAO/Abe.cpp:2557]: But I'll do it

**Pull request merged**